base/util/src/main/java/com/netscape/cmsutil/crypto/CryptoUtil.java

```diff
@@ -1611,6 +1611,41 @@ public static PKCS10 createCertificationRequest(String subjectName,
         return extn;
     }
 
+    /*
+     * addCTpoison adds the Certificate Transparency V1 poison extension
+     * to the Ceritificate Info
+     *
+     * @param certinfo X509CertInfo where the poison extension is to be added
+     *
+     * cfu
+     */
+    public static final String CT_POISON_OID = "1.3.6.1.4.1.11129.2.4.3";
+    public static final boolean CT_POISON_CRITICAL = true;
+    public static final byte CT_POISON_DATA[] =  new byte[] { 0x05, 0x00 };
+
+    public static void addCTpoison(X509CertInfo certinfo)
+                throws Exception {
+        String method = "CryptoUtil:addCTpoison: ";
+        ObjectIdentifier ct_poison_oid = new ObjectIdentifier(CT_POISON_OID);
+        Extension ct_poison_ext = null;
+        CertificateExtensions exts =  null;
+
```

Comment:
we should handle known exceptions (`CertificateException` and `IOException`) here:

````
 try {
            exts = (CertificateExtensions)
                    certinfo.get(X509CertInfo.EXTENSIONS);

            if (exts == null)
                throw new X509ExtensionException(method + " X509CertInfo.EXTENSIONS not found inf cetinfo");
            DerOutputStream out = new DerOutputStream();

            out.putOctetString(CT_POISON_DATA);

            ct_poison_ext = new Extension(ct_poison_oid, CT_POISON_CRITICAL, out.toByteArray());

            certinfo.set(X509CertInfo.EXTENSIONS, exts);
            exts.set(CT_POISON_OID, ct_poison_ext);
            certinfo.delete(X509CertInfo.EXTENSIONS);
            certinfo.set(X509CertInfo.EXTENSIONS, exts);
        } catch (CertificateException | IOException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
````

Reply:
right.  I also have 2nd thought about putting this in CryptoUtil.  I'll find a better place to land this. thanks.